### DISCUSSION TITLE
make easier to switch playing speed

### DISCUSSION THREAD

**Discussion by voidplayer** [libre-tube/LibreTube#5398] [Ideas]
### Describe your suggested feature

#5346 is out of scope. I wonder if its possible to modify the interface a lil so they are not 7 taps away as switching speed is a very common use case

Ideas:
* add 1.75x button as is very common speed
* letting you select the speed after clicking the video. This will make every video 2 taps instead of 1, and can be added as a option setting
* show default speed buttons directly after you tap the show options on the top right corner
* show a speed bar you can drag as direct option of the video

I can probably come up with more ideas to this issue if none of this are suitable :)

Thank you for your consideration and thanks for this great program!

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
> letting you select the speed after clicking the video. This will make every video 2 taps instead of 1, and can be added as a option setting
>
> show a speed bar you can drag as direct option of the video
>
This has already been somewhat rejected in https://github.com/libre-tube/LibreTube/issues/3023 because we don't really have much space there and don't want to clutter the UI with additional buttons in the player.

> show default speed buttons directly after you tap the show options on the top right corner
>
I don't think that's really a good idea, then there would be less space for the other options. Every user uses some options more frequently and some options less frequently. Some users might want to access the playback speed quicker, some might want to access the quality option quicker, some that like captions would like to enable these by one single click, ...
There's always going to be someone that's not happy with it, so I'd prefer to keep all options of the exact same importance by putting them into the same menu.

> add 1.75x button as is very common speed
>
I don't mind doing that if you think it's useful.

**Reply by clicky6** [reply]
How about giving options in settings to add extra buttons on the player UI like quick toggling of captions, video quality, speed so that users will decide for themselves what they need and so on & also giving options for users to remove existing buttons which are there.

**Comment by voidplayer** [comment]
> I don't think that's really a good idea, then there would be less space for the other options. Every user uses some options more frequently and some options less frequently

I understand every user has different preferences and i dont have the numbers, but i feel the speed one is a very common use case compared with the rest

Ankidroid is an example of showing a custom bar. All settings have an option to always show, never show and show if theres room. I feel libre-tube would benefit greatly of this kind of setup

> I don't mind doing that if you think it's useful.

I think is very useful thank you :)

firefox sound player also skip 1.75x for some reason feels very random as is as useful as the rest of the settings

**Reply by clicky6** [reply]
I agree, speed is very common for everyone,
Although For me personally quick toggling of captions is also very useful
And quick toggling of quality is not at all useful as one can set a preferred video quality resolution for all videos.

**Comment by mkdot7** [comment]
Playback speed control button would be super useful I used it frequently in ReVance. We should have option like Ankidroid as mentioned above or mentioned by clicky6.

> How about giving options in settings to add extra buttons on the player UI like quick toggling of captions, video quality, speed so that users will decide for themselves what they need and so on & also giving options for users to remove existing buttons which are there.
